src/Discord.Net.Commands/Info/ParameterInfo.cs

```diff
@@ -59,10 +59,23 @@ internal ParameterInfo(ParameterBuilder builder, CommandInfo command, CommandSer
         public async Task<TypeReaderResult> ParseAsync(ICommandContext context, string input, IServiceProvider services = null)
         {
             services = services ?? EmptyServiceProvider.Instance;
-            return await _reader.ReadAsync(context, input, services).ConfigureAwait(false);
+            var failedResults = new List<TypeReaderResult>();
```

Comment:
I'm not sure that this is the best approach. Due to the asynchronous nature of TypeReaders, we could be performing many expensive operations per typereader per parameter.

Reply:
Maybe you're right. I guess the alternative (and maybe more robust, but likely more complex way) would be some additional metadata/key. The question is: "how will the `CommandService` know which extra key to use?"